verl/single_controller/base/worker.py

```diff
@@ -148,12 +148,6 @@ def __init__(self, cuda_visible_devices=None) -> None:
             self.local_rank = int(os.environ['LOCAL_RANK'])
         ###
 
-        ###
-        # [SUPPORT AMD: torch]
-        if "AMD" in torch.cuda.get_device_name():
```

Comment:
`TaskRunner` is a cpu only actor used for RL algorithm flow control, while `Worker` is used for specific models like Actor/Critic/Reward/Reference, etc.  I think the `torch.cuda.set_device` is need here for AMD GPU to work properly.